rtlib/src/DisegniSegmenter.cpp

```diff
@@ -0,0 +1,167 @@
+    // It is needed for findContours()
+    Mat dist_8u;
+    dist.convertTo(dist_8u, CV_8U);
+    // Find total markers
+    std::vector<std::vector<Point>> contours;
+    findContours(dist_8u, contours, RETR_EXTERNAL, CHAIN_APPROX_SIMPLE);
+    // Create the marker image for the watershed algorithm
+    Mat markers = Mat::zeros(dist.size(), CV_32S);
+    // Draw the foreground markers
+    for (size_t i = 0; i < contours.size(); i++) {
+        drawContours(
+            markers, contours, static_cast<int>(i),
+            Scalar(static_cast<int>(i) + 1), -1);
+    }
+    // Draw the background marker
+    circle(markers, Point(5, 5), 3, Scalar(255), -1);
+    markers.convertTo(markers, CV_8U);
+    markers.convertTo(markers, CV_32S);
+
+    // Perform the watershed algorithm
+    watershed(imgResult, markers);
+    Mat mark;
+    markers.convertTo(mark, CV_8U);
+    bitwise_not(mark, mark);
+
+    // Returns the watershedded Mat image as distinct pixel values
+    return markers;
+}
+
+std::vector<Mat> DisegniSegmenter::split_merge_(Mat watershededImage)
+{
+    // Finding Boundary Boxes
+    struct MinMaxIdentifier {
+        cv::Point min{INT_MAXI, INT_MAXI};
+        cv::Point max{INT_MINI, INT_MINI};
+    };
+
+    // Finding Maximum and Minimum of a subregion
+    std::map<int32_t, MinMaxIdentifier> extremeFinder;
+    for (int y = 0; y < watershededImage.rows; y++) {
+        for (int x = 0; x < watershededImage.cols; x++) {
+
+            // Get label
+            auto label = watershededImage.at<int32_t>(y, x);
+
+            // Labels that are skipped because they aren't a part of the
+            // subregion
+            if (label == -1 || label == 255) {
+                continue;
+            }
+
+            // Make a new minmax if we don't have one for this label
+            if (extremeFinder.count(label) == 0) {
+                extremeFinder[label] = MinMaxIdentifier();
+                continue;
+            }
+
+            // This finds the max (top-left point) and min (bottom-right point)
+            // for each subregion
+            if (x < extremeFinder[label].min.x) {
+                extremeFinder[label].min.x = x;
+            };
+            if (y < extremeFinder[label].min.y) {
+                extremeFinder[label].min.y = y;
+            };
+            if (x > extremeFinder[label].max.x) {
+                extremeFinder[label].max.x = x;
+            };
+            if (y > extremeFinder[label].max.y) {
+                extremeFinder[label].max.y = y;
+            };
+        }
+    }
+
+    // Region of Interest taking Minimum and Maximum points
+    for (const auto& i : extremeFinder) {
+        int height = ((i.second.max.y) - (i.second.min.y));
+        int width = ((i.second.max.x) - (i.second.min.x));
+        Rect roi(i.second.min.x, i.second.min.y, width, height);
+        Mat image_roi = inputImage_(roi).clone();
+        resultImages_.push_back(image_roi);
```

Comment:
Bug: You need to clear `resultImages_` before this loop. Otherwise, the returned vector will keep getting larger and larger if you call `compute` multiple times.